react-native-in-app-payments-quickstart/app/screens/HomeScreen.tsx

```diff
@@ -147,57 +120,56 @@ export default class HomeScreen extends Component {
         console.log(ex);
       }
     }
-
-    this.setState({
-      canUseDigitalWallet: digitalWalletEnabled,
-    });
+    setcanUseDigitalWallet(digitalWalletEnabled)
   }
 
-  async onApplePayRequestNonceSuccess(cardDetails) {
-    if (this.chargeServerHostIsSet()) {
+
+  const onApplePayRequestNonceSuccess = async (cardDetails: any) => {
+    if (chargeServerHostIsSet()) {
       try {
         await chargeCardNonce(cardDetails.nonce);
         await SQIPApplePay.completeApplePayAuthorization(true);
-        this.setState({ applePayState: applePayStatus.succeeded });
-      } catch (error) {
+        setapplePayState(applePayStatus.succeeded)
+      } catch (error: any) {
         await SQIPApplePay.completeApplePayAuthorization(false, error.message);
-        this.setState({ applePayError: error.message });
+        setapplePayError(error.message)
       }
     } else {
       await SQIPApplePay.completeApplePayAuthorization(true);
-      this.setState({ applePayState: applePayStatus.nonceNotCharged });
+      setapplePayState(applePayStatus.nonceNotCharged)
       printCurlCommand(cardDetails.nonce, SQUARE_APP_ID);
     }
   }
 
-  async onApplePayRequestNonceFailure(errorInfo) {
+  const onApplePayRequestNonceFailure = async (errorInfo: any) => {
+    errorMsg = errorInfo.message;
     await SQIPApplePay.completeApplePayAuthorization(false, errorInfo.message);
-    showAlert('Error processing Apple Pay payment', errorInfo.message);
+    showAlert('Error processing Apple Pay payment', errorMsg);
   }
 
-  async onApplePayComplete() {
-    if (this.state.applePayState === applePayStatus.succeeded) {
+  const onApplePayComplete = async () => {
+    if (applePayState === applePayStatus.succeeded) {
       showAlert('Your order was successful',
         'Go to your Square dashboard to see this order reflected in the sales tab.');
-    } else if (this.state.applePayState === applePayStatus.nonceNotCharged) {
+    } else if (applePayState === applePayStatus.nonceNotCharged) {
       showAlert(
         'Nonce generated but not charged',
         'Check your console for a CURL command to charge the nonce, or replace CHARGE_SERVER_HOST with your server host.',
       );
-    } else if (this.state.applePayError != null) {
-      showAlert('Error processing Apple Pay payment', this.state.applePayError);
+    } else if (applePayError != null) {
+      showAlert('Error processing Apple Pay payment', applePayError);
     } else { // the state is none, so they canceled
-      this.showOrderScreen();
+      showOrderScreen();
     }
   }
 
-  async onGooglePayRequestNonceSuccess(cardDetails) {
-    if (this.chargeServerHostIsSet()) {
+  const onGooglePayRequestNonceSuccess = async (cardDetails: any) => {
```

Comment:
please use a concrete type instead of any

Reply:
Resolved it.